junit-jupiter-api/src/main/java/org/junit/jupiter/api/AssertTimeout.java

```diff
@@ -141,8 +148,11 @@ static void assertTimeoutPreemptively(Duration timeout, Executable executable, S
 				return future.get(timeoutInMillis, TimeUnit.MILLISECONDS);
 			}
 			catch (TimeoutException ex) {
+				ExecutionTimeoutException exception = new ExecutionTimeoutException("Execution timed out");
```

Comment:
Definitely we need a better message. Also, maybe we should use `cause` as variable name instead of `exception`. 🤔

Reply:
I think it would be helpful to include the thread name in the message: "Execution timed out in thread ...".